build: Update bazel to 3.4.1 and k/repo-infra to v0.2.0

mikedanese opened the pull request:
build: Update bazel to 3.4.1 and k/repo-infra to v0.2.0
Needed for Golang 1.16 updates: https://github.com/kubernetes/release/issues/1834

/kind cleanup
/sig release

```release-note
NONE
```

mikedanese committed:
bump bazel to 3.4.1

Change-Id: I7ae75acba5518782a99ede2d896a787238d7b744

mikedanese commented:
/retest

mikedanese commented:
/test pull-kubernetes-bazel-test-canary

mikedanese commented:
/test pull-kubernetes-bazel-test-canary

mikedanese commented:
/test pull-kubernetes-bazel-test-canary

mikedanese commented:
/retest

mikedanese commented:
I don't think we need this until after code freeze, but I'll let @justaugustus say when.

cc @BenTheElder @fejta

dims commented:
/retitle WIP: bump bazel to 3.4.1

mikedanese commented:
Oops, forgot to push.

mikedanese commented:
/retest

justaugustus reviewed (changes requested):
Chatted with @mikedanese and we'll want to first:
- bump the bazel version in k/repo-infra
- update any required deps in k/repo-infra
- cut a new k/repo-infra release
- target the new k/repo-infra in this PR
- play whack-a-mole with any failures
- ~curl up in a ball and cry~
- ...
- PROFIT

An additional vector is the go1.16 release that is "imminent", so overall, I think I'd prefer for this to wait until 1.21.

/hold
/milestone v1.21
/area release-eng
cc: @kubernetes/release-engineering

dims commented:
/uncc

mikedanese commented:
Are we ready?

/hold cancel

mikedanese commented:
Sorry, just reread you comment after vacation. We can tie this up with a repo-infra version bump if that's desirable.

MushuEE commented:
@amwat you might have some insight as to why kubetest is failing on bazel bump

mikedanese commented:
/retest

Looks like it failed due to flaky networking in the build cluster.

mikedanese committed:
bump repo-infra to 0.2.0

Change-Id: Id4fa13ca3cf092867bb0973294605962a8e5019c

mikedanese commented:
/retest

mikedanese commented:
/retest

BenTheElder commented:
looks to me like the bazel sandbox is newly interfering with unix sockets skimming https://prow.k8s.io/view/gs/kubernetes-jenkins/pr-logs/pull/96011/pull-kubernetes-bazel-test/1346729597261058048

mikedanese commented:
Yup, appears to be something in rules_go. I'll investigate.

mikedanese commented:
Here's the issue: https://github.com/bazelbuild/rules_go/issues/2776
And a fix: https://github.com/bazelbuild/rules_go/pull/2777

mikedanese committed:
fixup fixture logic

There was a bug with init() but it was resolved in
https://github.com/bazelbuild/rules_go/pull/2696

Changed to match other fixture methods.

Change-Id: I882b8535e5c5c117fb10c41d34c8eed1ccdb74bb

mikedanese commented on staging/src/k8s.io/legacy-cloud-providers/vsphere/vclib/fixtures/fixtures.go:
```diff
@@ -38,28 +38,43 @@ var (
+//
+// The approach taken here works for both go test and bazel on the assumption
+// that if and only if trimpath is passed, we are running under bazel.
+func pkgPath() (string, error) {
```
This is copied from https://github.com/kubernetes/kubernetes/pull/90872/

mikedanese commented:
/retest

mikedanese commented:
For staging/src/k8s.io/legacy-cloud-providers/vsphere/vclib/fixtures/fixtures.go change:

/assign @andrewsykim

justaugustus commented:
/lgtm
/approve

dims commented:
/approve

(For code organization / dependency updates only) :)

dims commented:
/hold

( for @andrewsykim )

andrewsykim reviewed (commented):
lgtm for `staging/src/k8s.io/legacy-cloud-providers/vsphere/vclib/fixtures/fixtures.go`

justaugustus commented:
/hold cancel

justaugustus commented:
/retest

justaugustus commented:
Awesome work, @mikedanese!!!
